### PR TITLE
Jetpack Social: Add Core Data model for PublicizeInfo

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #20806

> **Note**:  This PR targets 22.8 in case there's not enough time to merge this before the 22.7 branch is cut on Monday.

As titled, this adds the Core Data model for `PublicizeInfo`, which is a direct mapping of `RemotePublicizeInfo`. I'm intentionally isolating this PR since it contains quite a lot of changes.

## To test

There's nothing to test as of now, but please verify if there are any configurations that I've missed:

- The current model version should be set to 151.
- The one-to-one relationship between `Blog` and `PublicizeInfo` is optional for both ways.
- All the properties of `PublicizeInfo` are not optional, but the value defaults to `0`.
- The property mappings in `PublicizeInfo+CoreDataProperties` matches the model definition in the xcdatamodel.
- Check the additions in `MIGRATIONS.md` and verify that all the changes are covered and correct.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Model is unused, and feature is hidden behind a flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A.

3. What automated tests I added (or what prevented me from doing so)
N/A.

## PR submission checklist

N/A. This PR doesn't contain any UI changes.

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents [hunk 1]
```diff
@@ -0,0 +1,1039 @@
...
+    <entity name="PublicizeConnectionStatsRecordValue" representedClassName=".PublicizeConnectionStatsRecordValue" parentEntity="StatsRecordValue" syncable="YES">
+        <attribute name="followersCount" attributeType="Integer 64" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="iconURLString" optional="YES" attributeType="String" syncable="YES"/>
+        <attribute name="name" attributeType="String" syncable="YES"/>
+    </entity>
+    <entity name="PublicizeInfo" representedClassName=".PublicizeInfo" syncable="YES">
+        <attribute name="sharedPostsCount" optional="YES" attributeType="Integer 64" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="shareLimit" optional="YES" attributeType="Integer 64" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="sharesRemaining" attributeType="Integer 64" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="toBePublicizedCount" optional="YES" attributeType="Integer 64" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
```
Comment (wargcm) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents]: Should the `optional` field on `sharedPostsCount`, `shareLimit`, and `toBePublicizedCount` be removed here if they're required? Or maybe the migration notes need to be updated to match these along with the object?

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+        <relationship name="blog" optional="YES" maxCount="1" deletionRule="Nullify" destinationEntity="Blog" inverseName="publicizeInfo" inverseEntity="Blog"/>
```
Comment (wargcm) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents:674]: The `blog` property here on the `PublicizeInfo` object is `To One` currently but the migration notes state it should be `To Many`. Should one of these be updated to match the other?

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents [hunk 3]
```diff
@@ -0,0 +1,1039 @@
...
+        <relationship name="menuLocations" optional="YES" toMany="YES" deletionRule="Cascade" ordered="YES" destinationEntity="MenuLocation" inverseName="blog" inverseEntity="MenuLocation" syncable="YES"/>
+        <relationship name="menus" optional="YES" toMany="YES" deletionRule="Cascade" ordered="YES" destinationEntity="Menu" inverseName="blog" inverseEntity="Menu" syncable="YES"/>
+        <relationship name="pageTemplateCategories" optional="YES" toMany="YES" deletionRule="Cascade" destinationEntity="PageTemplateCategory" inverseName="blog" inverseEntity="PageTemplateCategory" syncable="YES"/>
+        <relationship name="posts" optional="YES" toMany="YES" deletionRule="Cascade" destinationEntity="AbstractPost" inverseName="blog" inverseEntity="AbstractPost"/>
+        <relationship name="postTypes" optional="YES" toMany="YES" deletionRule="Cascade" destinationEntity="PostType" inverseName="blog" inverseEntity="PostType" syncable="YES"/>
+        <relationship name="publicizeInfo" optional="YES" maxCount="1" deletionRule="Cascade" destinationEntity="PublicizeInfo" inverseName="blog" inverseEntity="PublicizeInfo"/>
```
Comment (wargcm) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents:201]: The migration notes state this should be required but it is marked `optional` here. Should one of these be updated to match the other?

**Review by wargcm — COMMENTED** [review]
There's a few things mismatching between the migration notes and the model. Could you update one of the two so they match?

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -9296,6 +9300,9 @@
...
 		FE18495727F5ACBA00D26879 /* DashboardPromptsCardCell.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DashboardPromptsCardCell.swift; sourceTree = "<group>"; };
+		FE1E200F2A45ACE900CE7C90 /* WordPress 151.xcdatamodel */ = {isa = PBXFileReference; lastKnownFileType = wrapper.xcdatamodel; path = "WordPress 151.xcdatamodel"; sourceTree = "<group>"; };
+		FE1E20132A47042400CE7C90 /* PublicizeInfo+CoreDataClass.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; name = "PublicizeInfo+CoreDataClass.swift"; path = "/Users/dvdchr/code/github/wpios/WordPress/Classes/Models/PublicizeInfo+CoreDataClass.swift"; sourceTree = "<absolute>"; };
+		FE1E20142A47042500CE7C90 /* PublicizeInfo+CoreDataProperties.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; name = "PublicizeInfo+CoreDataProperties.swift"; path = "/Users/dvdchr/code/github/wpios/WordPress/Classes/Models/PublicizeInfo+CoreDataProperties.swift"; sourceTree = "<absolute>"; };
```
Comment (wargcm) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: The absolute paths here are breaking the build.

Comment (dvdchr) [reply WordPress/WordPress.xcodeproj/project.pbxproj]: Ah, thanks for catching that. This is now fixed to be a relative path via https://github.com/wordpress-mobile/WordPress-iOS/pull/20946/commits/cc116c96e4f77da13c238495ba5f1090953c7442.

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents (hunk 3 above)
Comment (dvdchr) [reply WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents:201]: Thanks. All the relationships should be optional. The MIGRATIONS.md entry is now fixed via https://github.com/wordpress-mobile/WordPress-iOS/pull/20946/commits/38cfe772d63c46bb4a4884ef78b415e4862f2576.

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents (hunk 2 above)
Comment (dvdchr) [reply WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents:674]: It should be `To One`. Fixed via https://github.com/wordpress-mobile/WordPress-iOS/pull/20946/commits/38cfe772d63c46bb4a4884ef78b415e4862f2576. Thanks!

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents (hunk 1 above)
Comment (dvdchr) [reply WordPress/Classes/WordPress.xcdatamodeld/WordPress 151.xcdatamodel/contents]: Good catch! Yes, it should be removed — fixed them via https://github.com/wordpress-mobile/WordPress-iOS/pull/20946/commits/38cfe772d63c46bb4a4884ef78b415e4862f2576.

**Comment by dvdchr** [comment]
Ready for your second round of review @wargcm. Thanks!

**Review by wargcm — APPROVED** [review]
@dvdchr Everything looks good now. Thanks for making those changes! :shipit:

**Pull request merged**